### ISSUE TITLE
Some questions about training

### ISSUE THREAD

**Issue by SecretSCI** [limuhit/360-Image-Compression#2]
Dear Professor:
      I executed four ‘run_XX.sh' files in the order you said, and everything went smoothly.  But when I used the ’lic360demo.py' file to test the PSNR， I found that although BPP is far apart, the overall PSNR indicators are not different, all around 28. 
     Why does this happen? Is there any other operation required？

**Comment by limuhit** [comment]
See https://github.com/limuhit/360-Image-Compression/issues/1

**Comment by limuhit** [comment]
By the way, you may test the ’lic360demo.py' with the pre-trained models given in "https://github.com/limuhit/360-Image-Compression/blob/main/test/pretrained_models.md" and put the parameters under the directory described in  ’lic360demo.py'. If everything works fine, then you have correctly configured the environment.  After that, try to train your model.

**Comment by SecretSCI** [comment]
Thanks for your replying.
I have tested the lic360demo.py' with the pre-trained models. The Final PSNR is quite good. 
So I checked the 'train/run_XX_XX.sh', there is some question in my mind:
    - Run the sh file with the order in #1 , 9 checkpoints will be created in the end. But checkpoint is different with pre-trained models, which only include 'rt = 1, 0.6, 0.4'
    -  I test the checkpoint 'low_imp_ent_8_1000_61_61_0_1' with test dataset, It show 'Bitrate:0.557bpp, PSNR:28.81dB, SSIM:0.8443'. 
       But pretrained pth model is '0.720bpp, PSNR:34.04dB, SSIM:0.9331'. Why is the gap so big?   Is anything else should do in sh file and py file?  Because I just only run the 4 sh file in the train dir.

**Comment by limuhit** [comment]
There are several steps. The first step is to initialize the model parameter for three bitrate ranges, i.e., high-bitrate, medial-bitrate, and low-bitrate ranges.  Then, you can use the three initialization to target different bitrates in that range.  In the second step, we perform end-to-end optimization for the model. You must load the initialization parameter pre-trained in the first step to initialize the model. Try to check whether you have successfully loaded the pre-trained models. Or just start with a new random initialization. I guess you meet the second condition.  By the way, the sh files are just demos to show how to run those py files. To get the best parameters, you may further fine-turn the parameters. There is a trick, try with the highest bitrate model with the best performance, then gradually increase the parameter for the entropy loss to get a smaller bitrate. Usually, you will get a satisfactory model.

**Comment by limuhit** [comment]
By the way, there would be a step 0 here.  I find that in trainIMPDPP.py, there is a line for initialization "pdict = torch.load('./save_models/init/high_ssim_2.pt',map_location=device)". You may further pre-train a model with only the encoder, decoder, and quantization to produce the "high_ssim_2.pt".  There is a suggestion, it is equal to remove the importance map from the CMP_BASE model.

**Comment by limuhit** [comment]
By the way, there would be a step 0 here.  I find that in trainIMPDPP.py, there is a line for initialization "pdict = torch.load('./save_models/init/high_ssim_2.pt',map_location=device)". You may further pre-train a model with only the encoder, decoder, and quantization to produce the "high_ssim_2.pt".  There is a suggestion, it is equal to remove the importance map from the CMP_BASE model.
